docs/sdks/server-sdks/ruby.md

```diff
@@ -84,71 +138,152 @@ The SDK will invoke the `log_assignment` function with an `assignment` object th
-get_numeric_assignment(...)
-get_json_string_assignment(...)
-get_parsed_json_assignment(...)
-```
+### B. Automatically Updating the SDK Configuration
+
+After initialization, the SDK continues polling Eppo's API at 30-second intervals. This retrieves the most recent flag and experiment configurations such as variation values, targeting rules, and traffic allocation. This happens independently of assignment calls.
 
-### Handling `nil`
+:::note
 
-We recommend always handling the `nil` case in your code. Here are some examples of when the SDK returns `nil`:
+Changes made to experiments on Eppo's web interface are almost instantly propagated through our Content-delivery network (CDN) Fastly. Because of the refresh rate, it may take up to 30 seconds (± 5 seconds fuzzing) for those to be reflected by the SDK assignments.
 
-1. The **Traffic Exposure** setting on experiments/allocations determines the percentage of subjects the SDK will assign to that experiment/allocation. For example, if Traffic Exposure is 25%, the SDK will assign a variation for 25% of subjects and `nil` for the remaining 75% (unless the subject is part of an allow list).
+:::
 
-2. Assignments occur within the environments of feature flags. You must enable the environment corresponding to the feature flag's allocation in the user interface before `getStringAssignment` returns variations. It will return `nil` if the environment is not enabled.
 
-![Toggle to enable environment](/img/feature-flagging/enable-environment.png)
+:::info
 
-3. If `get_string_assignment` is invoked before the SDK has finished initializing, the SDK may not have access to the most recent experiment configurations. In this case, the SDK will assign a variation based on any previously downloaded experiment configurations stored in local storage, or return `nil` if no configurations have been downloaded.
+By default, the Eppo client initialization is asynchronous to ensure no critical code paths are blocked. For more information on handling non-blocking initialization, see our [documentation here](/sdks/common-issues#3-not-handling-non-blocking-initialization).
 
-### Debugging `nil`
+:::
 
-If you need more visibility into why `get_string_assignment` is returning `nil`, you can change the logging level to `Logger::DEBUG` to see more details in the standard output.
+## 4. Contextual Bandits
+
+To leverage Eppo's contextual bandits using the Ruby SDK, there are two additional steps over regular feature flags:
+1. Add a bandit action logger to the assignment logger
+2. Querying the bandit for an action
+
+### A. Add a bandit action logger to the assignment logger
+
+In order for the bandit to learn an optimized policy, we need to capture and log the bandit actions.
+This requires adding a bandit action logging callback to the AssignmentLogger class
+```ruby
+class MyLogger < EppoClient::AssignmentLogger
+    def log_assignment(assignment):
+        ...
+
+    def log_bandit_action(self, bandit_action):
```

Comment:
why the `self` argument?

In other SDKs and their documentation, we call the parameters `assignmentEvent` and `banditEvent`, as they contain more than just the assignment and action respectively.

Reply:
maybe a python-ism?

Reply:
Thanks for noticing - sorry I copied this from python and missed this.